persistence/test/setup_test.go

```diff
@@ -40,38 +39,62 @@ var (
 	DefaultUnstakingHeight = int64(-1)
 )
 
+var testPersistenceModule modules.PersistenceModule
+
 // See https://github.com/ory/dockertest as reference for the template of this code
 // Postgres example can be found here: https://github.com/ory/dockertest/blob/v3/examples/PostgreSQL.md
 func TestMain(m *testing.M) {
-	pool, resource := sharedTest.SetupPostgresDocker()
-	testPersistenceModule = sharedTest.PersistenceModule
-	testPostgresDB = sharedTest.PostgresDB
+	pool, resource, persistenceMod := sharedTest.SetupPostgresDockerPersistenceMod()
+	testPersistenceModule = persistenceMod
 	m.Run()
 	sharedTest.CleanupPostgresDocker(m, pool, resource)
 }
 
-var (
-	testPersistenceModule modules.PersistenceModule
-	testPostgresDB        *persistence.PostgresDB
-)
+func NewTestPostgresContext(t *testing.T, height int64) *persistence.PostgresContext {
+	ctx, err := testPersistenceModule.NewRWContext(height)
+	require.NoError(t, err)
+	db := &persistence.PostgresContext{
+		Height: height,
+		DB:     ctx.(persistence.PostgresContext).DB,
+	}
+	t.Cleanup(func() {
+		require.NoError(t, db.Release())
+		// DISCUSS_IN_THIS_COMMIT: Do we need this given that `Release` reverses the changes made to the database?
```

Comment:
I think it's cleaner and safer IMO to restart the whole stack. But not too strong on it

Reply:
I'm going to remove it so that it's easier for us to "run into errors" as we add more changes with unexpected circumstances.